src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -559,3 +559,89 @@ def _patch(self, crd, crd_name, cr_name, func):
 
         return "Success"
+
+    def get_ceph_image(self):
+        image = "ceph/ceph:v14.2.6"
+
+        try:
+            api_response = self.coreV1_api.list_namespaced_pod(self.rook_env.namespace, label_selector="app=rook-ceph-mon", timeout_seconds=10)
+            if api_response.items:
+                image = api_response.items[-1].spec.containers[0].image
+        except ApiException as e:
+            log.warning("Error getting ceph image: {}, using <{}> as default".format(e, image))
+
+        return image
+
+    def blink_light(self, ident_fault, on, locs):
+        # type: (str, bool, List[DeviceLightLoc]) -> string
+        operation_id = str(hash(locs))
+        message = ""
+
+        # job definition
+        job_metadata = client.V1ObjectMeta(name=operation_id, namespace= self.rook_env.namespace, labels={"ident": operation_id})
+        pod_metadata = client.V1ObjectMeta(labels={"ident": operation_id})
+        pod_container = client.V1Container(name="ceph-lsmcli-command",
+                                           image=self.get_ceph_image(),
+                                           command=["lsmcli",],
+                                           args=['local-disk-%s-led-%s' % (ident_fault,'on' if on else 'off'),
+                                                 '--path', locs.path or locs.dev,])
+        pod_spec = client.V1PodSpec(containers=[pod_container],
+                                    active_deadline_seconds=30, # Max time to terminate pod
+                                    restart_policy="Never",
+                                    node_selector= {"kubernetes.io/hostname": locs.host})
+        pod_template = client.V1PodTemplateSpec(metadata=pod_metadata,
+                                                  spec=pod_spec)
+        job_spec = client.V1JobSpec(active_deadline_seconds=60, # Max time to terminate job
+                                    ttl_seconds_after_finished=10, # Alfa. Lifetime after finishing (either Complete or Failed)
+                                    backoff_limit=0,
+                                    template=pod_template)
+        job = client.V1Job(api_version="batch/v1",
+                           kind="Job",
+                           metadata=job_metadata,
+                           spec=job_spec)
+
+        # delete previous job if it exists
+        try:
+            try:
+                api_response = self.batchV1_api.delete_namespaced_job(operation_id, self.rook_env.namespace, propagation_policy="Background")
+            except ApiException as e:
+                if e.status != 404: # No problem if the job does not exist
+                    raise
+
+            # wait until the job is not present
+            deleted = False
+            while not deleted:
+                api_response = self.batchV1_api.list_namespaced_job(self.rook_env.namespace, label_selector="ident=%s" % operation_id, timeout_seconds=10)
+                deleted = not api_response.items
```

Comment:
do we need a timeout?

Reply:
yes... better be nice with the CPU.

Reply:
Hm. What about if, of some unknown reason, the resource is never getting deleted?

Reply:
if we reach this line.. the delete of the job has been launched without any error, so k8s is just doing the task. If the k8s cannot finish with the task then what we have is a bug in k8s .... but apart of that. We have a second safety mechanism, that is that the job execution is limited to 60 seconds, and after finished (fail or succes) the job must be automatically deleted in 10 seconds. 
Again If k8s cannot delete the job we will have a bug in k8s ...

If we abort without deleting the job. (because the k8s bug), we are going to move the problem to other place , because if a job for a disk cannot be deleted .. then the creation of a new job for that disk is not going to be possible, and then we will have another different error caused because the problem of deletion.

Maybe what we can do is to add a retry-limit checking if the job has been deleted and abort if we reach this limit.. what do you think?

Reply:
sounds good! The problem with endless loops is, the orch cli hangs indefinitely making it much harder to understand the underlying problem.

Reply:
Done!